proptest/src/arbitrary/primitives.rs

```diff
@@ -44,3 +47,62 @@ mod test {
         u8 => u8, u16 => u16, u32 => u32, u64 => u64, u128 => u128
     );
 }
+
+#[cfg(all(feature = "unstable", feature = "never-arbitrary"))]
+mod never_arbitrary_tests {
+    use super::*;
+    use crate::prelude::*;
+
+    #[derive(Clone)]
+    enum Wrapper<T: Clone> {
+        A(T),
+        B,
+        C,
+    }
+
+    impl<T: Clone> std::fmt::Debug for Wrapper<T> {
+        fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+            match self {
+                Self::A(_) => write!(f, "Wrapper::A(_)"),
+                Self::B => write!(f, "Wrapper::B"),
+                Self::C => write!(f, "Wrapper::C"),
+            }
+        }
+    }
+
+    impl<T: Arbitrary + Clone + 'static> Arbitrary for Wrapper<T> {
+        type Parameters = T::Parameters;
+        type Strategy = BoxedStrategy<Self>;
+    
+        fn arbitrary_with(args: Self::Parameters) -> Self::Strategy {
+            prop_oneof![
+                any::<T>().prop_map(Wrapper::A),
+                Just(Wrapper::B),
+                Just(Wrapper::C),
+            ].boxed()
+        }
+    }
+    
+    impl Arbitrary for Wrapper<!> {
```

Comment:
unfortunately, due to coherence rules, this won't be possible to implement if `Wrapper` is defined in another crate as the rust trait coherence rules considers it to be a conflicting implementation.

Without this PR the error looks like this
```
19 | impl<T: Arbitrary> Arbitrary for Wrapper<T> {
   | ------------------------------------------- first implementation here
...
28 | impl Arbitrary for Wrapper<!> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Wrapper<!>`
   |
   = note: upstream crates may add a new impl of
trait `_IMPL_ARBITRARY_FOR_Wrapper::_proptest::arbitrary::Arbitrary` for type `!` in future versions
```

With this PR it still doesn't work, and the error looks like this
```
error[E0119]: conflicting implementations of trait `_IMPL_ARBITRARY_FOR_Wrapper::_proptest::arbitrary::Arbitrary` for type `Wrapper<!>`
  --> src/main.rs:13:10
   |
13 | #[derive(proptest_derive::Arbitrary, Clone, Debug)]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Wrapper<!>`
...
32 | impl Arbitrary for Wrapper<!> {
   | ----------------------------- first implementation here
   |
   = note: this error originates in the derive macro
`proptest_derive::Arbitrary`
(in Nightly builds, run with -Z macro-backtrace for more info)
```


There is an initiative inside of rustc to integrate negative impls into trait coherence rules, but until then this will be insufficient.
https://github.com/rust-lang/negative-impls-initiative/issues/1


I originally thought this would be a strict improvement, but actually judging from the error messages (IMO they end up regressing) - I think this is probably premature and should wait until the negative impls initiative furthers.

Perhaps hiding it behind another feature flag like you did is a good way to mitigate this concern!